### PR TITLE
Adding audit conformance tests suite

### PR DESCRIPTION
**Pull request by soltysh**
@stevekuznetsov this PR adds the additional conformance tests suite with audit on, we've talked about last week. Once this lands will you be able to add it to the nightly tests you've mentioned we have?

### CODE REVIEW

**Comment by soltysh** [comment]
[test]

File: test/extended/audit.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/test/extended/setup.sh"
+cd "${OS_ROOT}"
+
+os::test::extended::setup
+# enable auditing
+cp ${MASTER_CONFIG_DIR}/master-config.yaml ${MASTER_CONFIG_DIR}/master-config.orig.yaml
```
Comment (stevekuznetsov) [inline test/extended/audit.sh]: I think convention is `file.ext` to become `file.ext.old` or `file.ext.bak`

File: test/extended/setup.sh [hunk 2]
```diff
@@ -16,7 +16,40 @@ function os::test::extended::focus {
...
+  serial_only=( "${SERIAL_TESTS[@]}" )
+  serial_exclude=( "${EXCLUDED_TESTS[@]}" )
+
+  pf=$(join '|' "${parallel_only[@]}")
+  ps=$(join '|' "${parallel_exclude[@]}")
+  sf=$(join '|' "${serial_only[@]}")
+  ss=$(join '|' "${serial_exclude[@]}")
+
+  echo "[INFO] Running the following tests:"
+  TEST_REPORT_DIR= TEST_OUTPUT_QUIET=true ${EXTENDEDTEST} "--ginkgo.focus=${pf}" "--ginkgo.skip=${ps}" --ginkgo.dryRun --ginkgo.noColor | grep ok | grep -v skip | cut -c 20- | sort
```
Comment (stevekuznetsov) [inline test/extended/setup.sh]: While someone's mucking with this we need to either:
 - remove this
 - make it handle the case where no tests are selected
 - make it sane code and not rely on a 10x pipe so above step is easier

File: test/extended/setup.sh [hunk 3]
```diff
@@ -98,16 +131,27 @@ function os::test::extended::setup {
...
+    # to store the images created when the extended tests run
       sed -i 's/perFSGroup: null/perFSGroup: 896Mi/' $NODE_CONFIG_DIR/node-config.yaml
     fi
     echo "[INFO] Using VOLUME_DIR=${VOLUME_DIR}"
 
-    # This is a bit hacky, but set the pod gc threshold appropriately for the garbage_collector test.
-    os::util::sed 's/\(controllerArguments:\ \)null/\1\n    terminated-pod-gc-threshold: ["100"]/' \
-      ${MASTER_CONFIG_DIR}/master-config.yaml
+    # Update pod gc threshold appropriately for the garbage_collector test.
+    cp ${MASTER_CONFIG_DIR}/master-config.yaml ${MASTER_CONFIG_DIR}/master-config.orig.yaml
```
Comment (stevekuznetsov) [inline test/extended/setup.sh]: This is unrelated to audit, right? Maybe should be separate commit.

Also, unless we *want* to keep both versions of this config, clean up the useless one.

**Comment by soltysh** [comment]
[extended:audit]

**Comment by stevekuznetsov** [comment]
@soltysh cool!

**Comment by stevekuznetsov** [comment]
@soltysh I removed your previous test tags... to get just the one you want:

[testonlyextended][extended:audit]

**Comment by stevekuznetsov** [comment]
also to ensure conformance keeps running

[test]

**Comment by stevekuznetsov** [comment]
@soltysh is this still being pursued?

**Comment by soltysh** [comment]
I need to return to it, we should run those tests. I'll add to to X-mas todo list :wink

**Comment by soltysh** [comment]
I don't think this is needed anymore, since we're running our extended tests [with audit](https://github.com/openshift/origin/blob/f6171087ab255995d39d974e22f11c24ee6d3b93/test/extended/setup.sh#L108-L112) turned on.
